src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Tasks/DoNotUseWhenAllOrWaitAllWithSingleArgumentFixer.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Tasks
+{
+    public abstract class DoNotUseWhenAllOrWaitAllWithSingleArgumentFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+            DoNotUseWhenAllOrWaitAllWithSingleArgument.WaitAllRule.Id,
+            DoNotUseWhenAllOrWaitAllWithSingleArgument.WhenAllRule.Id);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var nodeToFix = root.FindNode(context.Span, getInnermostNodeForTie: true);
+
+            var semanticModel = await context.Document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var operation = (IInvocationOperation)semanticModel.GetOperation(nodeToFix, cancellationToken);
+            if (operation.TargetMethod.Name == nameof(Task.WaitAll))
+            {
+                var title = MicrosoftNetCoreAnalyzersResources.DoNotUseWaitAllWithSingleTaskFix;
+                context.RegisterCodeFix(new MyCodeAction(title,
+                    async ct =>
+                    {
+                        var editor = await DocumentEditor.CreateAsync(context.Document, ct).ConfigureAwait(false);
+                        editor.ReplaceNode(nodeToFix,
+                            editor.Generator.InvocationExpression(
+                                editor.Generator.MemberAccessExpression(
+                                    GetSingleArgumentSyntax(operation),
+                                    nameof(Task.Wait))
+                                ).WithTriviaFrom(nodeToFix)
+                            );
+
+                        return editor.GetChangedDocument();
+                    },
+                    equivalenceKey: title),
+                context.Diagnostics);
+            }
+            else if (!IsValueStored(operation))
```

Comment:
In this branch we're expecting `operation.TargetMethod.Name` to be "WaitAll"?

If so, consider an assertion

Reply:
It is, and assertion added locally. Draft PR isn't done yet.